sys/shell/commands/sc_nanocoap_vfs.c

```diff
@@ -137,3 +137,36 @@ int _nanocoap_get_handler(int argc, char **argv)
     }
     return res;
 }
+
+int _nanocoap_put_handler(int argc, char **argv)
+{
+    int res;
+    char *file, *url;
+    char buffer[CONFIG_NANOCOAP_QS_MAX];
+    char work_buf[coap_szx2size(CONFIG_NANOCOAP_BLOCKSIZE_DEFAULT) + 1];
```

Comment:
It doesn't [look like it](https://godbolt.org/z/G1h6qjWG6) but for a shell command I wouldn't worry about that either way.

Reply:
I think you are both right. I think this formally is a VLA, but since the compiler knows the exact size of the VLA at compile time, it will realize it as a regular array.

You could use `DECLARE_CONSTANT()` and later use that constant to also formally have this a non-VLA. But I see no benefit in this, as the machine code will be the exact some anyway. Maybe unless we add `-Wvla` to the compiler flags, in which case `CECLARE_CONSTANT()` would be the way to go.